//     ___         __  ____                   
//    / __/_ ____ / /_/ / /_ _   _______  ___    C++ for Fast LLM Inference
//   / _/ _ `(_-</ __/ / /  ' \_/ __/ _ \/ _ \   version 0.0.1
//  /_/ \_,_/___/\__/_/_/_/_/_(_)__/ .__/ .__/   https://github.com/thansen0
//                                /_/  /_/    
//
//  SPDX-FileCopyrightText: 2025 thansen0 <https://github.com/thansen0>
//  SPDX-License-Identifier: Unlicense

syntax = "proto3";

package llm_request;

// saves in same file as /protos, new name creates new folder for compiled code
// option go_package = "/";

service AskLLMQuestion {
    rpc PromptLLM (LLMInit) returns (LLMInference) {}
}

message LLMInit {
    string apiKey = 1;
    string prompt = 2;
}


message LLMInference {
    string answer = 1;
}

